src/esp/gfx/PbrShader.cpp

```diff
@@ -43,39 +43,14 @@ namespace gfx {
 namespace {
 enum TextureUnit : uint8_t {
   BaseColor = 0,
-  Roughness = 1,
-  Metallic = 2,
-  Normal = 3,
-  // NoneRoughnessMetallic or OcclusionRoughnessMetallic texture
-  Packed = 4,
-  Emissive = 5,
+  MetallicRoughness = 1,
+  Normal = 2,
+  Emissive = 3,
 };
 }  // namespace
 
-PbrShader::Flags PbrShader::generateCorrectFlags(Flags originalFlags) {
-  Flags result = originalFlags;
-  // NOTE:
-  // The priority of different kind of textures is as follows (priority
-  // means if two textures with different priorities exist at the same time,
-  // shader will adopt the texture with the higher priority, and ignore the
-  // other one.)
-  // 0 (highest): OcclusionRoughnessMetallicTexture
-  // 1          : NoneRoughnessMetallicTexture
-  // 2          : RoughnessTexture, MetallicTexture
-  if (result & Flag::OcclusionRoughnessMetallicTexture) {
-    result &= ~Flag::NoneRoughnessMetallicTexture;
-    result &= ~Flag::RoughnessTexture;
-    result &= ~Flag::MetallicTexture;
-  } else if (result & Flag::NoneRoughnessMetallicTexture) {
-    result &= ~Flag::RoughnessTexture;
-    result &= ~Flag::MetallicTexture;
-  }
```

Comment:
Just in case you forgot all our previous private or public discussions -- the MaterialData APIs in Magnum were not built in a generic way because I like to be an architecture astronaut designing overly generic APIs or because I did not read the glTF spec carefully enough. It's because, even if we ignore other file formats, glTF allows for much more flexibility beyond what the core spec says:

- [MSFT_packing_occlusionRoughnessMetallic](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Vendor/MSFT_packing_occlusionRoughnessMetallic/README.md) (three new packing modes)
- [MSFT_packing_normalRoughnessMetallic](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Vendor/MSFT_packing_normalRoughnessMetallic/README.md) (another packing variant)
- And [several other proposals](https://github.com/KhronosGroup/glTF/issues/1456) appearing, mainly useful for furniture models where you often have tiled textures where all maps don't necessarily have the same transformation.

I mean, there's nothing wrong with supporting just the core glTF files and failing import for everything else, but why do that **now**, after we both spent _so much time_ discussing and implementing all the generic code you just removed?

Reply:
- [MSFT_packing_occlusionRoughnessMetallic](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Vendor/MSFT_packing_occlusionRoughnessMetallic/README.md) (three new packing modes)
Not a problem with this PR. The occlusion will be in the `MetallicRoughness` texture. I already considered this situation see the `Flag` in the `PbrShader`. I have `PackedOcclusionTexture` to handle this case and `SeparateOcclusionTexture` to handle the other one.
- [MSFT_packing_normalRoughnessMetallic](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Vendor/MSFT_packing_normalRoughnessMetallic/README.md) (another packing variant)
Good to know. Even without this PR, the current code cannot handle this extension. So anything regarding this extension is totally a different topic, has nothing to do with the changes of this PR.

- [several other proposals](https://github.com/KhronosGroup/glTF/issues/1456)
This is a proposal from 2018 that is not finished yet. This proposal aims to apply the transforms per texture (per channel, because roughness, metallic are in different channels within the same image), not per image. Again, this is a totally separate topic, has nothing to do with this PR, has no effects on current glTF Spec yet (and may not happen in near future.)

> I mean, there's nothing wrong with supporting just the core glTF files and failing import for everything else.

To clarify, as long as we are talking about glTF, including the extensions you mentioned, this PR will NOT fail except that [MSFT_packing_normalRoughnessMetallic](https://github.com/KhronosGroup/glTF/blob/master/extensions/2.0/Vendor/MSFT_packing_normalRoughnessMetallic/README.md) (another packing variant), which cannot be handled by the original code before this PR as well.

Reply:
(Marking as resolved based on the previous discussions, either private or via the weekly.)